src/inference/include/openvino/runtime/intel_npu/level_zero/level_zero.hpp

```diff
@@ -0,0 +1,144 @@
+ * which is shared with LevelZero context object.
+ * The plugin object derived from this class can be obtained either with
+ * CompiledModel::get_context() or Core::create_context() calls.
+ * @ingroup ov_runtime_level_zero_npu_cpp_api
+ */
+class ZeroContext : public RemoteContext {
+protected:
+    /**
+     * @brief NPU device name
+     */
+    static constexpr const char* device_name = "NPU";
+
+    /**
+     * @brief Default constructor which can be used in derived classes to avoid multiple create_context() calls
+     */
+    ZeroContext() = default;
+
+public:
+    // Needed to make create_tensor overloads from base class visible for user
+    using RemoteContext::create_tensor;
+
+    /**
+     * @brief Constructs context object from user-supplied LevelZero context handle
+     * @param core A reference to OpenVINO Runtime Core object
+     */
+    ZeroContext(Core& core) {
+        *this = core.get_default_context(device_name).as<ZeroContext>();
+    }
+
+    /**
+     * @brief Returns the underlying LevelZero context handle.
+     * @return `void*`
+     */
+    void* get() {
+        return get_params().at(ov::intel_npu::l0_context.name()).as<void*>();
+    }
+
+    /**
+     * @brief This function is used to obtain remote tensor object from user-supplied Direct3D 12 Core object
+     * @param type Tensor element type
+     * @param shape Tensor shape
+     * @param buffer A void* object that should be wrapped by a remote tensor
+     * @return A remote tensor instance
+     */
+    ZeroBufferTensor create_tensor(const element::Type type, const Shape& shape, void* buffer) {
+        AnyMap params = {{ov::intel_npu::mem_type.name(), ov::intel_npu::MemType::SHARED_BUF},
+                         {ov::intel_npu::mem_handle.name(), buffer}};
+        return create_tensor(type, shape, params).as<ZeroBufferTensor>();
+    }
+
+    /**
+     * @brief This function is used to obtain remote tensor object from user-supplied DMA-BUF System Heap object
+     * @param type Tensor element type
+     * @param shape Tensor shape
+     * @param fd A int object that should be wrapped by a remote tensor
+     * @return A remote tensor instance
+     */
+    ZeroBufferTensor create_tensor(const element::Type type, const Shape& shape, int fd) {
+        AnyMap params = {{ov::intel_npu::mem_type.name(), ov::intel_npu::MemType::SHARED_BUF},
+                         {ov::intel_npu::mem_handle.name(), reinterpret_cast<void*>(static_cast<intptr_t>(fd))}};
+        return create_tensor(type, shape, params).as<ZeroBufferTensor>();
+    }
+
+    /**
+     * @brief This function is used to obtain remote tensor object
+     * @param type Tensor element type
+     * @param shape Tensor shape
+     * @param tensor_type Type of the tensor to be shared, input, output or binded
+     * @return A remote tensor instance
+     */
+    ZeroBufferTensor create_tensor(const element::Type type, const Shape& shape, const TensorType tensor_type) {
```

Comment:
In working on a POC code in ONNX RT using GPU remote tensors I had to use create_host_tensor. If not as part of this PR then afterwards I'd like to see that implemented for the NPU as well and mapped to shared host memory.

Reply:
changed the name of the method to create_l0_host_tensor

Reply:
Also, in a further PR will implement the create_host_tensor method on our end as well.